lib/mist/s3/buckets.py

```diff
@@ -37,21 +38,41 @@ def handle_delete(self):
         try:
             patterns = [re.compile(arg) for arg in args[1:]]
             buckets = self.find_matching(patterns)
-            bucket_names = [{"bucket": bucket.name} for bucket in buckets]
             try:
-                while len(buckets) > 0:
-                    bucket = buckets[0]
+                for bucket in buckets:
                     bucket.delete()
-                    buckets = buckets[1:]
             except Exception as e:
-                Logger.error("Error deleting S3 bucket '%s': %s" & (buckets[0].name, e))
+                Logger.error("Error deleting S3 bucket '%s': %s" % (buckets[0].name, e))
                 self.resp.send_error("Error deleting bucket '%s': %s" % (buckets[0].name, e))
             if len(buckets) == 0:
                 self.resp.append_body([], template="empty_result")
             else:
-                self.resp.append_body({"buckets": buckets}, template="delete_buckets")
+                bucket_names = [bucket.name for bucket in buckets]
+                self.resp.append_body({"buckets": ", ".join(bucket_names)}, template="delete_buckets")
```

Comment:
Yeah it'd be nice if we returned a list of bucket names like: `[{"bucket": "..."}, ...]`